### ISSUE TITLE
Feature request: Convert Video / Audio in the background

### ISSUE THREAD

**Issue by inglesuniversal** [Cactooz/FFmpegVideoConverter#1]
Hello, your script is amazing! Is there a way for the user to upload a file (e.g. VIDEO.mov) and then auto-convert to mp4 in the background and send an EMAIL or UPDATE database once process is complete?

I can take care of the notification part, but just want to know if running this in the background is nearly possible?

Thanks and #STAYSAFE

**Comment by Cactooz** [comment]
Hey, thanks :D
Yes, this is doable for both the conversion and the email.
For the conversion do you mean that it should just change the .mov directly into a .mp4 without adding/changing any tags, subtitles or cover art?  If so then I can implement a quite simple bulk conversion that converts all files in for example a folder from one file type to another, but without adding or changing any other data in the file.

**Comment by inglesuniversal** [comment]
Yes, the overall idea or main problem I am going through is with a few formats which are not totally compatible between apple and windows devices... e.g whatsapp audios are exported as .OGG and Videos are created on Macs and iPhones as MOV... So I am creating a cloud storage app that lacks that feature.... I am able to upload just about any type of file but OGGs are not playable on iPhone and MOV files won't play on Chrome Browser and I would love to be able to grab a single frame of the source (MOV) video and make a poster image based on the source video resolution. If you need more information, files and what have you... I can definitely help in every way possible.

Thanks so much and thanks for your willingness to help me out.

Hugo

**Comment by Cactooz** [comment]
I see your issue, I had the same one and that's why I made the script to begin with :D

You said you want to grab an frame in the full resolution and use that as an image. Do you want it as just a plain image file or do you want to attach it as artwork for the video/sound file itself?

**Comment by inglesuniversal** [comment]
Once again... Can't thank you enough on your interest and support to the issues I have described. I would say that I can have a /tmp folder in the LAMP server to run all this code and then.... dump a single JPG image from the video file with a text ending like -> VIDEO_NAME_thumb.jpg (based on the video's default resolution)... It would be cool to have a trigger that can run a custom API call to update my database and at the most, a second option to trigger an SMS or EMAIL notification... All those calls could run independently of the script just like an add-on feature to make our lives easier =).  I don't know if your script could cover an scenario if the first few frames of the source VIDEO are in plain black... it can skip them until a non-black frame is detected (just a wild thought).

#STAYSAFE and Have a great weekend!

**Comment by Cactooz** [comment]
Hey just a little update, I've started to plan out how to implement this feature with the rest of the program.
However, I'm going to be quite busy in the upcoming month so it might take a while before the feature is implemented. 

Thanks for your understanding :D

**Comment by inglesuniversal** [comment]
No problem on that... I know and understand that good things require time. Please keep me posted!

Good luck & fortune on ALL your projects!

**Comment by Cactooz** [comment]
I've implemented a bulk conversion feature now that allows you to convert multiple files at once. Use -b when you start the main file and then you can define which files you want to convert and into which file format they will get converted to.
I've not tested this fully yet but feel free to test it in its current state if you want to 😄

**Comment by inglesuniversal** [comment]
Can't wait to try it ... However, I do not see any instructions, just sh files. Can you provide a link or youtube video on how to use the current features? Thanks so much!

**Comment by Cactooz** [comment]
If you use ``./VideoConverter.sh -h`` then you will get a help instruction. Currently just use ``./VideoConverter.sh -b`` to start the bulk conversion feature and then follow the instructions on the screen.
I can make more instructions when the new feature is fully implemented.

**Comment by inglesuniversal** [comment]
Nice... nice... nice... I'm just starting to have fun with all this.... LOL

Is there a way to trigger a single or batch conversion from a PHP script?  Without any manual intervention so that users can get their files converted from MOV to MP4 as the upload them into a base folder

Thanks

**Comment by Cactooz** [comment]
No clue, I've never worked in .php. But it's probably possible if you Google.

**Comment by inglesuniversal** [comment]
I found an example and it is pretty easy to do.... Take a look...

`<?php
$contents = file_get_contents('./Ogg2Mp3.sh');
echo shell_exec($contents);`


NOTE: Be aware that folder must have 777 permission to be called from outside the server or part of an API component!

**Comment by inglesuniversal** [comment]
I will be sharing some code, so you can see how simple it is.

Question: Does your script require that the video is first downloaded locally so it can generate a single thumbnail from the video?

Can the video be stored elsewhere and still work?

By that I mean, if it is possible to generate a single thumbnail from a remote video without the need to download it into the server or script local path.


Regards

**Comment by Cactooz** [comment]
The script does only convert local files, it does not work with external files. It doesn't make any thumbnails of videos either, it just has an option to add custom artwork for the file. Normal thumbnails are generated by Windows (or your operating system).
Though it's possible with FFmpeg to extract a .jpg or .png image from one of the frames in a video. That is another possible script and future though, not connected to the bulk conversion feature.

I have no plan to integrate a PHP integration. Firstly because I have no PHP knowledge and secondly because the script is not really built to work on a website.
> NOTE: Be aware that folder must have 777 permission to be called from outside the server or part of an API component!

Changing permissions for a folder is risky and should be done manually, so the script doesn't have the responsibility for other things that might break.
